### PR TITLE
Add DVGeometryESP driver and test infrastructure

### PR DESCRIPTION
**Pull request by bbrelje**
## Purpose
This PR adds the DVGeometryESP driver and unit tests.

DVGeometryESP provides a way to optimize aerodynamic surfaces using geometry parameterized in the AFRL EngineeringSketchPad software

## Type of change
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
A simple cube test case with analytic derivatives is provided, along with a NACA 0012 airfoil case which provides a more realistic setting. This branch can successfully converge the singlepoint NACA 0012 airfoil opt problem in the public MACH tutorial and I plan to add that example to the MACH tutorial repo once this is merged.

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [X] I have run unit and regression tests which pass locally with my changes
- [X] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Comment by bbrelje** [comment]
I need to roll the version of EngineeringSketchPad in the docker image. It ended up being easier than expected to convert to the new, fully-supported pyOCSM API so I no longer need to pull ESP and pyOCSM from a sketch gdrive location. However, it will probably fail tests on the first try

**Comment by bbrelje** [comment]
We can't merge this until the Docker deps image is updated with ESPbeta. The relevant tests were all skipped in this build

**Comment by bbrelje** [comment]
executes order #66

**Comment by eirikurj** [comment]
@bbrelje new images are currently running

**Comment by joanibal** [comment]
It looks good to me. 

Just fix the failing formatting tests

**Review by joanibal — DISMISSED** [review]

**Comment by marcomangano** [comment]
@joanibal I am pretty sure pyGeo is not linted and reformatted yet, I would leave this for a separate PR once we have merged these new features

File: tests/reg_tests/test_DVGeometryESP.py
Comment (nwu63) [inline tests/reg_tests/test_DVGeometryESP.py:2]: This is going to raise an error if `mpi4py` is not available. Consider wrapping with try/except as you did with pyOCSM.

**Comment by nwu63** [comment]
All the ESP tests were skipped since the dependency was not available in the deps image yet. This should be ready in a few hours.

But I also noticed that several pyGeo tests were skipped, does anyone know why?
```
The following tests were skipped:
test_DVConstraints.py:RegTestPyGeo.test_15
test_DVConstraints.py:RegTestPyGeo.test_16
test_DVConstraints.py:RegTestPyGeo.test_17
...ESP tests
```

**Comment by sseraj** [comment]
This PR is from a fork so it did not pull the private image. All the geograd tests were skipped as a result.

**Comment by marcomangano** [comment]
> But I also noticed that several pyGeo tests were skipped, does anyone know why?

@nwu63 they are skipped because they require `Geograd`. Probably we need the private image for that?

**Comment by nwu63** [comment]
Ah okay, we really need to start having descriptive test names... at a minimum I'd like to wait until we test the ESP tests before merging -- those dependencies will be in the public image.

**Comment by nwu63** [comment]
Everything passed! Except intel which is expected since ESP doesn't play nice with Intel compilers

File: tests/reg_tests/test_DVGeometryESP.py
Comment (bbrelje) [reply tests/reg_tests/test_DVGeometryESP.py]: Fixed!

**Review by nwu63 — APPROVED** [review]

**Review by joanibal — APPROVED** [review]

**Comment by joanibal** [comment]
> at a minimum I'd like to wait until we test the ESP tests before merging

@nwu63 when will this happen?

**Comment by nwu63** [comment]
> @nwu63 when will this happen?

Now :) ESP tests ran and passed on Azure.

**Pull request merged**
